politeiawww/comments.go

```diff
@@ -5,10 +5,95 @@
 package main
 
 import (
+	"github.com/decred/politeia/plugins/comments"
+	pi "github.com/decred/politeia/politeiawww/api/pi/v1"
 	www "github.com/decred/politeia/politeiawww/api/www/v1"
 	"github.com/decred/politeia/politeiawww/user"
+	"github.com/google/uuid"
 )
 
+func convertCommentsPluginPropStateFromPi(s pi.PropStateT) comments.StateT {
+	switch s {
+	case pi.PropStateUnvetted:
+		return comments.StateUnvetted
+	case pi.PropStateVetted:
+		return comments.StateVetted
+	}
+	return comments.StateInvalid
+}
+
+func (p *politeiawww) processComments(c pi.Comments) (*pi.CommentsReply, error) {
+	log.Tracef("processComments: %v", c.Token)
+
+	// Call comments plugin to get comments
+	reply, err := p.comments(comments.GetAll{
+		Token: c.Token,
+		State: convertCommentsPluginPropStateFromPi(c.State),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var cr pi.CommentsReply
+	// Transalte comments
+	cs := make([]pi.Comment, 0, len(reply.Comments))
+	for _, cm := range reply.Comments {
+		// Get comment's author username
+		// Parse string uuid
+		uuid, err := uuid.Parse(cm.UUID)
+		if err != nil {
+			return nil, err
+		}
+		// Get user
+		u, err := p.db.UserGetById(uuid)
+		if err != nil {
+			return nil, err
+		}
+		cs = append(cs, pi.Comment{
```

Comment:
Put the conversion in a conversion function.  This won't be the only time you'll need to convert between these types.